src/nvim/msgpack_rpc/channel.c

```diff
@@ -312,6 +315,9 @@ bool channel_close(uint64_t id)
   }
 
   close_channel(channel);
+  if (!kv_size(channel->call_stack)) {
```

Comment:
Well, yes. In fact I had done it that way the first time. I did that optimization later, as at that point, we know channel is not closed, so we can save checking one condition (and the function call).
But I have no problem in changing it back to previous situation, if its considered better from a readability point of view.